alluxio-ui/master/src/containers/pages/Browse/Browse.tsx

```diff
@@ -0,0 +1,277 @@
+import {faFile, faFolder} from '@fortawesome/free-solid-svg-icons'
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
+import React from 'react';
+import {connect} from 'react-redux';
+import {Link} from 'react-router-dom';
+import {Alert, Button, Form, FormGroup, Input, Label, Table} from 'reactstrap';
+import {Dispatch} from 'redux';
+
+import {FileView, Paginator} from '@alluxio/common-ui/src/components';
+import {IFileBlockInfo, IFileInfo} from '@alluxio/common-ui/src/constants';
+import {createDebouncedFunction, parseQuerystring} from '@alluxio/common-ui/src/utilities';
+import {IApplicationState} from '../../../store';
+import {fetchRequest} from '../../../store/browse/actions';
+import {IBrowse} from '../../../store/browse/types';
+
+import './Browse.css';
+
+interface IPropsFromState {
+  browse: IBrowse;
+  errors: string;
+  loading: boolean;
+  location: {
+    search: string;
+  };
+}
+
+interface IPropsFromDispatch {
+  fetchRequest: typeof fetchRequest;
+}
+
+interface IBrowseProps {
+  refreshValue: boolean;
+}
+
+interface IBrowseState {
+  end?: string;
+  limit?: string;
+  offset?: string;
+  path?: string;
+  lastFetched: {
+    end?: string;
+    limit?: string;
+    offset?: string;
+    path?: string;
+  };
+  textAreaHeight?: number;
+}
+
+type AllProps = IPropsFromState & IPropsFromDispatch & IBrowseProps;
+
+class Browse extends React.Component<AllProps, IBrowseState> {
+  private readonly textAreaResizeMs = 100;
+  private readonly debouncedUpdateTextAreaHeight = createDebouncedFunction(this.updateTextAreaHeight.bind(this), this.textAreaResizeMs, true);
+
+  constructor(props: AllProps) {
+    super(props);
+
+    const {path, offset, limit, end} = parseQuerystring(this.props.location.search);
+    this.state = {end, limit, offset, path: path || '/', lastFetched: {}};
+  }
+
+  public componentWillReceiveProps(props: AllProps) {
```

Comment:
It seems like this lifecycle is deprecated?

Reply:
great catch, fixed.